src/Extensions/SwiftypeSiteTreeCrawlerExtension.php

```diff
@@ -171,4 +200,60 @@ protected function getLogger(): LoggerInterface
     {
         return Injector::inst()->get(LoggerInterface::class);
     }
+
+
+    public function setUrlsToCrawl(array $urls) {
+        $this->urlsToCrawl = $urls;
+    }
+
+    public function getUrlsToCrawl(): array
+    {
+        return $this->urlsToCrawl;
+    }
+
+    private function getOwnerKey(): string
+    {
+        $owner = $this->getOwner();
+        $key = $owner->ClassName . $owner->ID;
+        return $key;
+    }
+
+    /**
+     * collate Urls to crawl
+     *
+     * @return void
+     */
+    public function collateUrls(): void
+    {
+        $urls = $this->getUrlsToCrawl();
```

Comment:
I think we should probably(?) be clearing out any old URLs if we re-request to `collateUrls()`. So, I'm thinking `$urls = [];`.

Reply:
this applies to below but the `key` is there because the extension instance is a singleton and you might have a different page/owner on a separate run. This will clear out previous urls from the same owner...